lib/src/rules/dartdoc_summary_line.dart

```diff
@@ -0,0 +1,185 @@
+/// be found. Throws a [PermissionError] if the file is present but could
+/// not be deleted.
+void delete(String path) {
+  ...
+}
+```
+''';
+
+/// Enforces that DartDoc blocks start with a single, brief line that ends in
+/// a period.
+///
+/// (Like the one above!)
+class DartdocSummaryLine extends LintRule {
+  DartdocSummaryLine()
+      : super(
+            name: 'dartdoc_summary_line',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    var visitor = _Visitor(this);
+    registry.addComment(this, visitor);
+  }
+}
+
+class _Visitor extends SimpleAstVisitor<void> {
+  final LintRule rule;
+
+  _Visitor(this.rule);
+
+  /// The summary line should be 'brief' according to the Dart recommendations,
+  /// however 'brief' is not defined perfectly. Instead of enforcing a fixed
+  /// number of characters, we just look at how many lines are part of the
+  /// summary, and if we deem them to be too many (2 should be plenty), we fail
+  /// the lint
+  ///
+  /// The above summary fails with all the checks that this lint adds, it has
+  /// more than one sentence, is too long, and doesn't add a period at the end.
+  ///
+  /// It has it all!
+  @override
+  void visitComment(Comment node) {
+    if (!node.isDocumentation || !node.tokens.first.lexeme.startsWith('///')) {
+      // Ignore /** comments.
+      return;
+    }
+
+    Iterable<Token> summary = _getSummary(node);
+
+    if (!_endsWithPeriod(summary)) {
+      Token last = summary.last;
+      // Highlight the last character of the sentence...
+      rule.reportLintForOffset(last.offset + last.lexeme.length, 1);
+    }
+    int? sentenceBreakOffset = _findSentenceBreak(summary);
+    if (sentenceBreakOffset != null) {
+      // Highlight the position of the period that is breaking the summary.
+      rule.reportLintForOffset(sentenceBreakOffset, 1);
+    }
+    if (summary.length > maxSummaryLines) {
+      // Highlight the extra lines.
+      summary.skip(maxSummaryLines).forEach((token) {
+        rule.reportLintForOffset(token.offset, 3);
+      });
+    }
+  }
+}
+
+// Some utility functions
+
+/// Removes the first triple slash of a `line` and returns its trimmed value.
+String _trimSlashes(String line) => line.replaceFirst('///', '');
+
+/// Determines if a token `tk` represents an empty comment line.
+bool _isNotBlankLine(Token tk) => _trimSlashes(tk.lexeme).trim().isNotEmpty;
+
+/// Retrieves the summary line of a [Comment] node.
+///
+/// This will take "lines" (tokens or actual strings) from the beginning of the
+/// node until an empty line is found.
+Iterable<Token> _getSummary(Comment node) =>
+    node.tokens.takeWhile(_isNotBlankLine);
+
+/// Checks that the last of the `lines` ends in a period.
+bool _endsWithPeriod(Iterable<Token> tokens) =>
```

Comment:
I think we'll need to allow colons as well. For example: https://dart.dev/guides/language/effective-dart/documentation#prefer-backtick-fences-for-code-blocks